core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -383,60 +393,179 @@ object Battle {
+        }
+        // Remove improvements, add fallout
+        tile.improvement = null
+        tile.improvementInProgress = null
+        tile.turnsToImprovement = 0
+        tile.roadStatus = RoadStatus.None
+        if (tile.isLand && !tile.isImpassible() && !tile.terrainFeatures.contains("Fallout")) {
+            if (tile.terrainFeatures.any { attacker.getCivInfo().gameInfo.ruleSet.terrains[it]!!.uniques.contains("Resistant to nukes") }) {
+                if (Random().nextFloat() < 0.25f) {
+                    tile.terrainFeatures.removeAll { attacker.getCivInfo().gameInfo.ruleSet.terrains[it]!!.uniques.contains("Can be destroyed by nukes") }
+                    tile.terrainFeatures.add("Fallout")
+                }
+            } else if (Random().nextFloat() < 0.5f) {
+                tile.terrainFeatures.removeAll { attacker.getCivInfo().gameInfo.ruleSet.terrains[it]!!.uniques.contains("Can be destroyed by nukes") }
                 tile.terrainFeatures.add("Fallout")
+            }
         }
-
-        for (civ in attacker.getCivInfo().getKnownCivs()) {
-            civ.getDiplomacyManager(attackingCiv)
-                    .setModifier(DiplomaticModifiers.UsedNuclearWeapons, -50f)
+    }
+    
+    private fun nukeStrength2Effect(attacker: MapUnitCombatant, tile: TileInfo) {
+        // https://forums.civfanatics.com/threads/unit-guide-modern-future-units-g-k.429987/#2
+        // https://www.carlsguides.com/strategy/civilization5/units/aircraft-nukes.php
+        // Testing done by Ravignir
+        var damageModifierFromMissingResource = 1f
+        val civResources = attacker.getCivInfo().getCivResourcesByName()
+        for (resource in attacker.unit.baseUnit.getResourceRequirements().keys) {
+            if (civResources[resource]!! < 0 && !attacker.getCivInfo().isBarbarian())
+                damageModifierFromMissingResource *= 0.5f // I could not find a source for this number, but this felt about right
         }
-
-        // Instead of postBattleAction() just destroy the missile, all other functions are not relevant
-        if ((attacker as MapUnitCombatant).unit.hasUnique("Self-destructs when attacking")) {
-            attacker.unit.destroy()
+        // Damage and/or destroy cities
+        val city = tile.getCity()
+        if (city != null && city.location == tile.position) {
+            var populationLoss = 1f;
+            for (unique in city.civInfo.getMatchingUniques("Population loss from nuclear attacks -[]%")) {
+                populationLoss *= 1 - unique.params[0].toFloat() / 100f
+            }
+            if (city.population.population < 5) {
+                city.destroyCity()
+            } else {
+                city.population.population = (city.population.population * (0.6 + Random().nextFloat() * 0.2) * populationLoss).toInt()
+                if (city.population.population < 5 && populationLoss == 1f) city.population.population = 5
```

Comment:
This seems very odd. If I had 5 population, that was then decreased to 3, then you increase it back to 5? Why?

Reply:
Because on 
https://www.carlsguides.com/strategy/civilization5/units/aircraft-nukes.php
I found the following sentence:
> Bomb Shelters will not stop a Nuclear Missile's massive power from leveling a City unless it has 5 or higher Population. In that case, the population will not fall below 5. If the Population is 4, the City is gone.

So it seems to me that if you have a bomb shelter, your city should not go below 5 population. I did implement this wrong (it should be `populationLoss < 1f`), so there is something to change here, I do agree.

Reply:
K